src/catalog.rs

```diff
@@ -358,6 +385,30 @@ impl DefaultCatalog {
         (Arc::from(table_name.to_string()), Arc::new(table))
     }
 
+    fn build_table<'a, I>(
+        &self,
+        table_name: &str,
+        table_columns: I,
+    ) -> (Arc<str>, Arc<dyn TableProvider>)
+    where
+        I: Iterator<Item = &'a AllDatabaseColumnsResult>,
+    {
+        let table_columns_vec = table_columns.collect_vec();
+
+        // Build a delta table but don't load it yet; we'll do that only for tables that are
+        // actually referenced in a statement, via the async `table` method of the schema provider.
+
+        // TODO: if the table has no columns, the result set will be empty, so we use the default UUID (all zeros).
```

Comment:
Are there any implications? do we use this UUID to apply some DDL modifications to 0-column tables later down the line?

Reply:
Yes there are—we use this UUID to scope our root object store for a particular table by adding it to the end of the path (or in other words, prefixing all table paths with it, relative to the root store uri). 

I'll think about a way to better handle no-column situation.

Reply:
I realized that this comment is misleading—there's no way that the vec in `build_table` is empty since before we pass it down we group the original `collection_columns_vec` by table name (`.group_by(|col| &col.table_name)`), which means there was at least one element of the vector to begin with in order for grouping to work (likewise for `build_legacy_table`).

Moreover, in case of new tables, we only really need the uuid besides the name, which we can also fetch with the group by call as it's unique per table. In other words the rest of the `AllDatabaseColumnsResult` fields don't play a role in instantiating new tables, and I aim to simplify that loading logic after dropping the legacy tables completely.

On the other hand, this means that tables without columns wouldn't be loaded at all in the schema provider's map. However, this problem is averted since delta-rs doesn't allow column-less table creation (and we don't yet support `ALTER table DROP column`). Therefore, for now I'll just add a test to lock that in so it doesn't change underneath us.